src/misc/setup.cpp

```diff
@@ -989,6 +986,33 @@ bool Config::ParseConfigFile(char const * const configfilename) {
 	return true;
 }
 
+parse_environ_result_t parse_environ(const char * const * envp) noexcept
+{
+	std::list<std::tuple<std::string, std::string>> props_to_set;
+
```

Comment:
For duplicate or repeat settings, maybe we should use a std::set (or map) and not list?

>

Reply:
Edit: I now see this was only temporary and goes away in a later commit.

Reply:
1. No duplicate env variables are guaranteed because this is `environ`
2. We might have multiple values associated with every section… Right now code splits the env variable name DOSBOX_FOO_BAR=BAZ into section name FOO and setting BAR=BAZ, so I'm not sure how to fit that into a set or map
3. I do not like std::list, but in this case an average number of elements will be 0 - this feature is very rarely used. I think using list in this case is a bit more appropriate then deque (which would pre-allocate a buffer).

Reply:
Right, excellent point regarding environment already being unique keys!